### DISCUSSION TITLE
one-year-after-the-entity-framework-vote-of-no-confidence/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#34] [Blog comments]
# one-year-after-the-entity-framework-vote-of-no-confidence/

https://blogs.taiga.nl/martijn/posts/one-year-after-the-entity-framework-vote-of-no-confidence/

**Comment by disqus-comments-migrator** [comment]
<em>Adam D. commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 24 2009, 11:52</em>

---

<p>Even with good things like ASP.NET MVC, I go with Castle's Monorail instead. I totally agree that the marketing department at MS is doing a disservice to many people's careers.</p>

**Comment by disqus-comments-migrator** [comment]
<em>Eber Irigoyen commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 24 2009, 11:58</em>

---

<p>EF might not have tons of features, but from what I've read lately, performance is not one of its weaknesses, specially when compared to NHibernate, am I missing something?</p>

**Comment by disqus-comments-migrator** [comment]
<em>KristoferA commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2009, 07:23</em>

---

<p>I don't think you can blame Microsoft if end users make bad decisions and go with an inferior toolkit...</p><p>The flaws and shortcomings of EFv1 are well documented around the intarweb, so it should come as no surprise to anyone that it generates 'killer' SQL queries that can bring a db server onto its' knees in a pool of table scans.</p><p>Devs need to learn to evaluate the tools and technologies they are going to use, and choose the appropriate tool for the task _before_ going off developing software...</p><p>JMHO</p>

**Comment by disqus-comments-migrator** [comment]
<em>KristoferA commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2009, 07:30</em>

---

<p>@Eber It has a number of performance issues. If your model is bigger than northwind/adventureworks (i.e. &gt;100 tables or so) you'll notice a performance hit whenever it loads the 'model artifacts' (csdl/ssdl/msl) when you're about to hit the db. Next, many common and simple query constructs can result in terrible SQL queries with lots of redundant joins, where clause predicates that force table scans etc.</p><p>Funny, since they already had L2S to look at, and L2S generates excellent SQL queries that are well optimized at runtime to please the db server...   ...however, it seems like they decided to not look at any of the smart optimizations in L2S and just did their own thing without considering what would happen db-side.</p>

**Comment by disqus-comments-migrator** [comment]
<em>Danny Simmons commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2009, 09:49</em>

---

<p>I'm sorry that you have had frustrations with the first release of the EF.  May I humbly suggest that you give the next release of the EF (the one that will ship with .net 4 and is available in beta form now) a chance before completely writing off the EF.  It's certainly not perfect, but I hope you will find some significant improvements.</p><p>- Danny</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2009, 05:11</em>

---

<blockquote><br><strong><a href="#comment-304" rel="nofollow noopener" title="#comment-304">KristoferA</a> :</strong><br><p>I don’t think you can blame Microsoft if end users make bad decisions and go with an inferior toolkit…</p><br><p>The flaws and shortcomings of EFv1 are well documented around the intarweb, so it should come as no surprise to anyone that it generates ‘killer’ SQL queries that can bring a db server onto its’ knees in a pool of table scans.</p><br><p>Devs need to learn to evaluate the tools and technologies they are going to use, and choose the appropriate tool for the task _before_ going off developing software…</p><br></blockquote><p><br>Well, that's more or less the point I'm trying to make: it's pushed by MS so that developers are told by their management to use it. Managers don't read about the flaws of Version 1.0.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2009, 05:17</em>

---

<blockquote><br><strong><a href="#comment-306" rel="nofollow noopener" title="#comment-306">Danny Simmons</a> :</strong><br><p>I’m sorry that you have had frustrations with the first release of the EF.  May I humbly suggest that you give the next release of the EF (the one that will ship with .net 4 and is available in beta form now) a chance before completely writing off the EF.  It’s certainly not perfect, but I hope you will find some significant improvements.</p><br></blockquote><p><br>From what I've seen, the next version looks a whole lot better and I'll definitively will give it a good look. That aside, my frustrations are caused by the fact that the V1 version was positioned as the default way of doing data access and that many developers would be better off with an alternative solution that is available right now.</p>

**Comment by disqus-comments-migrator** [comment]
<em>KristoferA commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 25 2009, 08:54</em>

---

<blockquote><br><strong><a href="#comment-310" rel="nofollow noopener" title="#comment-310">martijn </a> :</strong>Well, that’s more or less the point I’m trying to make: it’s pushed by MS so that developers are told by their management to use it. Managers don’t read about the flaws of Version 1.0.</blockquote><p><br>hmm. those 'managers' would then fall under one or several of the following labels: incompetent (for not doing their due diligence, and for not listening to those who have), naïve (for taking marketing at face value -- do they also believe in laundry detergent TV commercials?), inexperienced (seen _any_ stable v1 product out of redmond? I haven't...).</p><p>oh well, it will be interesting to see what EFv3 brings, I expect it to be a lot better than v1...</p>

**Comment by disqus-comments-migrator** [comment]
<em>Bas commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at oktober 09 2009, 11:51</em>

---

<p>I love to be in control of my code.<br>EF will generate a lot of crap.<br>Look at Model.designer.cs it always makes me laugh:<br>//This code was generated by a tool.</p>
